client/files.go

```diff
@@ -0,0 +1,226 @@
+package client
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/beyondstorage/go-storage/v4/pairs"
+	"github.com/beyondstorage/go-storage/v4/services"
+	"github.com/beyondstorage/go-storage/v4/types"
+
+	"github.com/beyondstorage/beyond-ftp/utils"
+)
+
+func (c *Handler) handleSTOR() {
+	path := c.absPath(c.param)
+
+	c.storeOrAppend(path, false)
+}
+
+func (c *Handler) handleAPPE() {
+	_, ok := c.storager.(types.Appender)
+	if !ok {
+		c.WriteMessage(StatusCommandNotImplemented, "this type of storage is not support append")
+		return
+	}
+	path := c.absPath(c.param)
+	c.storeOrAppend(path, true)
+}
+
+func (c *Handler) storeOrAppend(path string, append bool) {
+	c.ctxRest = 0
+	tr, err := c.TransferOpen()
+	if err != nil {
+		c.WriteMessage(StatusCannotOpenDataConnection, err.Error())
+		return
+	}
+
+	if err := c.upload(path, tr, append); err != nil {
+		c.TransferClose()
+		c.WriteMessage(StatusFileActionNotTaken, err.Error())
+		return
+	}
+
+	select {
+	case <-c.commandAbortCtx.Done():
+		c.WriteMessage(StatusTransferAborted, "Connection closed; transfer aborted")
+	default:
+		c.TransferClose()
+		c.WriteMessage(StatusClosingDataConn, "transfer finished")
+	}
+}
+
+func (c *Handler) upload(path string, tr utils.Conn, append bool) error {
+	file := new(bytes.Buffer)
+	size, err := io.Copy(file, tr)
+	if err != nil {
+		return err
+	}
+
+	if appender, ok := c.storager.(types.Appender); ok {
+		object, err := c.storager.Stat(path)
+		if err != nil && !errors.Is(err, services.ErrObjectNotExist) {
+			return err
+		}
+
+		if !append || errors.Is(err, services.ErrObjectNotExist) {
+			object, err = appender.CreateAppendWithContext(c.commandAbortCtx, path)
+			if err != nil {
+				return err
+			}
+		}
+
+		_, err = appender.WriteAppendWithContext(c.commandAbortCtx, object, file, size)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	_, err = c.storager.WriteWithContext(c.commandAbortCtx, path, file, size)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Handler) handleRETR() {
+	defer func() {
+		c.ctxRest = 0
+	}()
+	path := c.absPath(c.param)
+	tr, err := c.TransferOpen()
+	if err != nil {
+		c.WriteMessage(StatusCannotOpenDataConnection, err.Error())
+		return
+	}
+
+	_, err = c.storager.ReadWithContext(c.commandAbortCtx, path, tr, pairs.WithOffset(c.ctxRest))
+	if err != nil {
+		c.TransferClose()
+		c.WriteMessage(StatusActionNotTaken, err.Error())
+		return
+	}
+
+	select {
+	case <-c.commandAbortCtx.Done():
+		c.WriteMessage(StatusTransferAborted, "Connection closed; transfer aborted")
+	default:
+		c.TransferClose()
+		c.WriteMessage(StatusClosingDataConn, "transfer finished")
+	}
+}
+
+func (c *Handler) handleCHMOD(params string) {
+
```

Comment:
write `not implemented` here?

Reply:
`site` command is removed because we now do not need it.